portfolio/src/main/webapp/script.js

```diff
@@ -28,103 +28,214 @@ function addRandomGreeting() {
 }
 
 //*********************** LOADING SCREEN ***********************
-function setVisible(selector, bool) {
+
+/**
+ * Sets elements to visible or not.
+ */
+function setHidden(selector, bool) {
     const el = document.getElementById(selector);
-    el.classList.toggle('hiddenclass', bool);
+    el.classList.toggle('hidden', bool);
 }
 
-function swapLoaderForContent() {
-    setVisible('header', false);
-    setVisible('content', false);
-    setVisible('loaderid', true);
+/**
+ * Gets rid of the loading screen and displays the page
+ */
+function swapLoaderForContent(){
+    setHidden('header', false);
+    setHidden('content', false);
+    setHidden('loaderid', true);
 }
 
+// env variables
+let listenNext = null;
+let retrievedInput = null;
+let match_index = 0;
+let matchingNodes = null;
+
 //*********************** SEARCH BAR FUNCTIONS ***********************
-/**
-  * Swaps the loader for the content and enables search
-  */
+
 window.onload = function(){
-    swapLoaderForContent();
+    swapLoaderForContent(); // displays page after loaded
+
+    // Next button event listener, always active
+    listenNext = document.getElementById("nextbutton");
+    listenNext.addEventListener("click", searchHandler);
+
+
+    //Runs search on enter
     document.getElementById('searchbar').onkeydown = function(e){
         if(e.keyCode == 13){
-            runSearch(document.getElementById('searchbar')
-                .value.toLocaleLowerCase());
+            // i only want single word alphanumerics to go through
+            retrievedInput = document.getElementById('searchbar').value;
+            const regex = new RegExp(/^[a-z0-9]+$/i);
+            if (regex.test(retrievedInput) === false){
+                shakeSearchBar();
+            } else {
+                searchSetup();
+            }
         }
     };
 }
 
+/**
+ * Handler function for the event listener
+ */
+function searchHandler(){
+    highlightHandler(matchingNodes, retrievedInput, match_index);
+}
+
+/**
+ * Code to shake the search bar and make it glow
+ * Calls a bunch of css transforms
+ */
+function shakeSearchBar(){
+    const el = document.getElementById('searchbar');
+    el.preventDefault;
+    el.classList.remove('searchbarglow')
+    void el.offsetWidth;
+    el.classList.add('searchbarglow');
+}
+
+
 //*********************** SEARCH IMPLEMENTATION ***********************
+
+
 /**
-  * Retreives the HTML from the source
-  */
-function retrieveHtml(){
-    var source = document.body.innerHTML;
-
-    if (source != null){
-        return source.toLocaleLowerCase();    //toLocaleLowerCase() to ignore caps
-    } else {
-        alert("source is null");
+ * Finds all the occurences of a substring in a string
+ */
+function findNeedleInHaystack(small, big) {
```

Comment:
small and big are not descriptive names. Add parameter definitions to your jsdoc:
https://jsdoc.app/about-getting-started.html

Also, I suggest using more descriptive names like:
```
function findSubstringIndices(substr, str) {
  ...
}
```